### PR TITLE
File permissions check documentation changes

### PR DESCRIPTION
**Pull request by akshya96**
https://hashicorp.atlassian.net/browse/VAULT-6037

### CODE REVIEW

**Review by HridoyRoy — APPROVED** [review]
lgtm!

**Review by HridoyRoy — APPROVED** [review]
lgtm!

**Review by ccapurso — APPROVED** [review]
Looks good!

File: website/content/docs/configuration/index.mdx [hunk 1]
```diff
@@ -9,9 +9,7 @@ description: Vault server configuration reference.
 Outside of development mode, Vault servers are configured using a file.
 The format of this file is [HCL](https://github.com/hashicorp/hcl) or JSON.
 
-By default, Vault expects the config directory and files to be owned by the
-user running Vault. It also expects no write or execute permissions for group or others.
-This check can be disabled via the environment variable `VAULT_DISABLE_FILE_PERMISSIONS_CHECK`.
+@include 'plugin-file-permissions-check.mdx'
```
Comment (fairclothjm) [inline website/content/docs/configuration/index.mdx:12]: Do we need this to be in this file twice? I see it is present on [line 123](https://github.com/hashicorp/vault/pull/15656/files#diff-93d96b9260b2330c857c3ff78bac8ab079a6c6b1e39d75d07d02ce9f0793b811R123)

Comment (akshya96) [reply website/content/docs/configuration/index.mdx:12]: Just wanted to give the users an idea about check file permissions for the config file. Further down, I have added it in places where we talk about plugins.

**Review by fairclothjm — APPROVED** [review]

**Pull request merged**